x-pack/test/security_solution_endpoint/services/endpoint.ts

```diff
@@ -164,30 +198,35 @@ export class EndpointTestResources extends FtrService {
           },
         };
 
-    // If we have a specific number of endpoint hosts to check for, then use that number,
-    // else we just want to make sure the index has data, thus just having one in the index will do
-    const size = ids.length || 1;
-
-    await this.retry.waitFor('endpoint hosts', async () => {
-      try {
-        const searchResponse = await this.esClient.search({
-          index: metadataCurrentIndexPattern,
-          size,
-          body,
-          rest_total_hits_as_int: true,
-        });
+    await this.waitForIndex(ids, metadataCurrentIndexPattern, body, timeout);
+  }
 
-        return searchResponse.hits.total === size;
-      } catch (error) {
-        // We ignore 404's (index might not exist)
-        if (error instanceof errors.ResponseError && error.statusCode === 404) {
-          return false;
+  async waitForUnitedEndpoints(ids: string[] = [], timeout = this.config.get('timeouts.waitFor')) {
+    const body = ids.length
+      ? {
+          query: {
+            bool: {
+              filter: [
+                {
+                  terms: {
+                    'agent.id': ids,
+                  },
+                },
+                // make sure that both endpoint and agent portions are populated
+                // since agent is likely to be populated first
+                { exists: { field: 'united.endpoint.agent.id' } },
+                { exists: { field: 'united.agent.agent.id' } },
+              ],
+            },
+          },
         }
+      : {
+          query: {
+            match_all: {},
```

Comment:
Wondering: in this case, maybe in order to speed the search up, we should limit the result to 1 only? we're just trying to ensure data exists right? (maybe do that change above too on line 197).

Reply:
that makes sense